### PR TITLE
Only show the Fraud & Risk tour if the user clicked on the "Learn more" button

### LINKED ISSUE #6887: Only show the Fraud & Risk tour if the user clicked on the "Learn more" button
### Description

We're currently showing the Fraud & Risk tour element when the user accesses the **Payments > Settings** page which is shown just the first time after completing the tour (by going through all steps or closing it). This behavior can be disruptive for the users if they aren't navigating from the Fraud & Risk banner.

This issue aims to update the flow required to render the tour element, which should be shown only if the user clicked on the "Learn more" button from the discoverability banner.

### Acceptance criteria

* The tour element should popup just if the user clicked on the "Learn more" button on the WCPay homepage
* Guarantee that the auto-scroll works correctly

### Dev notes

- We are currently appending a custom query parameter on the "Learn more" button ([GitHub](https://github.com/Automattic/woocommerce-payments/blob/develop/client/components/fraud-risk-tools-banner/components/banner-actions/index.tsx#L30)). We can use this mechanism to render the tour on the settings page.

### Additional context

- p1690511327086489-slack-C01B8KNUYSW

### PR DESCRIPTION
**Pull request by allie500**
Fixes #6887 

#### Changes proposed in this Pull Request

This PR adds a check for the custom query parameter we add to the WCPay settings URL when the Fraud and Risk Tools learn more button is clicked. If this query parameter is present and the other two checks pass, then the tour is shown, otherwise it isn't. This is done to prevent a sub-standard user experience in the event a merchant has not dismissed the tour yet, doesn't click 'learn more', and goes directly to the settings page where they are greeted by a grayed out, inactive screen for a second before the smooth scroll takes them to the bottom of the page for the tour. 

#### Testing instructions

- Checkout this PR branch
- Build the client: `npm run build:client`
- All tests should pass
- Delete FRT related options:
  - `npm run wp option delete wcpay_frt_discover_banner_settings`
  - `npm run wp option delete wcpay_fraud_protection_welcome_tour_dismissed`
- Visit the WCPay settings page: `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments`
- The welcome tour should not load.
- Visit the WCPay overview page: `/wp-admin/admin.php?page=wc-admin&path=/payments/overview`
- Click the learn more button on the FRT discoverability banner to go back to WCPay settings.
- Upon page load, the screen should turn gray and smooth scroll to the bottom of the settings page where the Fraud Protection section is. You should see the welcome tour card.
- Dismiss the tour.
- Go back to the WCPay overview page. Click the learn more button once more. 
- Upon page load the screen should remain active, the page should smooth scroll to the Fraud Protection section but the welcome tour should not re-appear.
- Start Jurassic Tube or Ngrok to access your local site via smartphone.
- Delete the welcome tour option again: `npm run wp option delete wcpay_fraud_protection_welcome_tour_dismissed`
- Login to your site admin via smartphone.
- Repeat the test steps above to confirm that the behavior is the same on a smartphone.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _[Release testing instructions](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-6.3.0#fraud-and-risk-tools-welcome-tour-display-updates)_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6896` or branch name `fix/6887-only-show-frt-tour-from-learn-more-button-click` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6887-only-show-frt-tour-from-learn-more-button-click) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: ad74e329bb89f6e089dd6367b1a4cc0408a450a8
- Build time: 2023-08-02 14:46:47 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/fraud-protection/tour/index.tsx [hunk 1]
```diff
@@ -45,15 +45,19 @@ const options = {
 const FraudProtectionTour: React.FC = () => {
 	const { isWelcomeTourDismissed } = wcpaySettings.fraudProtection;
 
+	const searchParams = new URLSearchParams( window.location.search );
```
Comment (eduardoumpierre) [inline client/settings/fraud-protection/tour/index.tsx:48]: **Optional:** What do you think about using the `@wordpress/url` dependency to get the query param value? We would have to import `import { getQueryArg } from '@wordpress/url';` and then change the code to something like:

```suggestion
	const anchorParam = getQueryArg( window.location.search, 'anchor' );
	const isTourParam = '#fp-settings' === anchorParam;
```

File: client/settings/fraud-protection/tour/index.tsx [hunk 2]
```diff
@@ -45,15 +45,19 @@ const options = {
...
+	const anchorParam = searchParams.get( 'anchor' );
+	const isTourParam = '#fp-settings' === anchorParam;
```
Comment (dwainm) [inline client/settings/fraud-protection/tour/index.tsx:50]: hasTourParam reads slightly better.

**Review by eduardoumpierre — APPROVED** [review]
The code is easy to understand, the tests are updated, and everything works as expected.

I left a comment about an optional change, so feel free to merge it without addressing it.

Thanks for working on this! 🚢

File: client/settings/fraud-protection/tour/index.tsx (hunk 2 above)
Comment (allie500) [reply client/settings/fraud-protection/tour/index.tsx:50]: Thanks for the suggestion. I chose to leave it as `isTourParam`. My thinking was that it is checking to see if the `anchorParam` **is** the Tour Param string we expect.

File: client/settings/fraud-protection/tour/index.tsx (hunk 1 above)
Comment (allie500) [reply client/settings/fraud-protection/tour/index.tsx:48]: Thanks for this suggestion. I like it, but I chose to leave it as is as I had seen `URLSearchParams` used in a few different areas of the codebase. Plus, I wanted to be sure to get this into `develop` before the code freeze tomorrow.

**Pull request merged**